source/build.py

```diff
@@ -0,0 +1,169 @@
+
+        for attr in attrs:
+            if ((
+                'ja' in self.pruneThisLang and
+                tag == 'a' and
+                attr[0] == 'href' and
+                ('/ja/' in attr[1] or attr[1].startswith('ja/'))
+            ) or (
+                'en' in self.pruneThisLang and
+                tag == 'a' and
+                attr[0] == 'href' and
+                ('/en/' in attr[1] or attr[1].startswith('en/'))
+            )):
+                #mark for removal: TOC items of the other language.
+                self.results.add(('prune', prunePos))
+
+                # retroactively prevent removal of main links from top level page to
+                # each language's index page.
+                if (
+                    not('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath) and
+                    self.targetFullPath.endswith('/index.html') and
+                    (attr[1] == 'en/' or attr[1] == 'ja/')
+                ):
+                    self.results.remove(('prune', prunePos))
+
+
+# ============================================================================
+
+
+if ('mkdocs.yml' in lsResults) and ('docs' in lsResults) and isMkdocsInstalled:
+    print('All dependencies present. Initiate build...')
+    subprocess.call('mkdocs build --clean', shell=True)
+    print('Mkdocs build successful.')
+    siteDirectory = cwd + "/site"
+    htmlFileFullPaths = set([])
+
+    print('Crawling files...')
+    for root, subDirectories, files in os.walk(siteDirectory):
+        for filename in os.listdir(root):
+            if filename.endswith('.html'):
+                htmlFileFullPaths.add(root + '/' + filename)
+                logging.debug(root + '/' + filename + ' found!')
+
+    for htmlFileFullPath in htmlFileFullPaths:
+        logging.debug('accessing ' + htmlFileFullPath)
+        readHtmlFile = open(htmlFileFullPath, 'r')
+        htmlString = readHtmlFile.read()
+        parser = PostMkdocsParser(htmlFileFullPath)
+        parser.feed(htmlString)
+
+        resultsSet = parser.results.copy()
+        readHtmlFile.close()
+        parser.__kill__()
+
+        if resultsSet:
+            htmlLines = htmlString.splitlines()
+            resultsList = []
+            for result in resultsSet:
+                resultsList.append(result)
+            resultsList.sort()
+            resultsList.reverse()
+
+            for result in resultsList:
+                position = int(result[1][0])
+
+                if result[0] == 'injectTagManager':
+                    for line in googleTagManagerScriptLines:
+                        htmlLines.insert(position, line)
+                        position += 1
+
+                if result[0] == 'prune':
+                    logging.debug('pop! goes ' + htmlLines[position-1])
+                    htmlLines.pop(position-1)
+
+            htmlNewString = ''
+            for line in htmlLines:
+                htmlNewString += (line + '\n')
```

Comment:
Python have this cool `join` function https://docs.python.org/2/library/string.html#string.join
check it out :sunglasses: